src/main/java/org/myrobotlab/service/Joystick.java

```diff
@@ -247,8 +252,54 @@ public void poll() {
   }
 
   public Joystick(String n, String id) {
-    super(n, id);
+    super(n, id);    
+  }
+
+  @Override
+  public void onInterfaceRegistered(String serviceName, String interfaceName) {
+    if (AnalogListener.class.toString().equals(interfaceName)) {
+      analogListenerServices.add(serviceName);
+      broadcastState();
+    }
+  }
+
+  @Override
+  public void onInterfaceReleased(String serviceName, String interfaceName) {
+    if (AnalogListener.class.toString().equals(interfaceName)) {
+      analogListenerServices.remove(serviceName);

```

Comment:
I feel like runtime should be owning the lists of all possible analog listener services.  i worry that when services are maintaining their own lists here, it could get out of sync from the actual list of services from runtime.

Reply:
runtime now owns the data structure - "requestedAttachMatrix"
its a definition of {Type} -> { interested itnerface } -> Set of services names that can provide it